src/Avalonia.Controls/Calendar/Calendar.cs

```diff
@@ -2128,5 +2168,124 @@ protected override void OnTemplateApplied(TemplateAppliedEventArgs e)
             KeyUp += Calendar_KeyUp;
         }
 
+        private sealed class CalendarBlackoutDatesCollectionInternal : CalendarBlackoutDatesCollection
```

Comment:
I can't understand what this class is doing. Can you explain?

Reply:
@grokys - The original implementation of `CalendarBlackoutDatesCollection` was copied from Silverlight and required a reference to the owning `Calendar` to create an instance. I split the code into two classes and moved everything that required the `Calendar` reference to an internal class allowing `CalendarBlackoutDatesCollection` to be instanced in ViewModels for binding purposes. The `Calendar` reference is used for validation purposes.

Reply:
I still don't understand this... For example, the internal collection calls `_owner.UpdateMonths()` when an item is added or removed. But this won't happen when the user assigns a non-internal instance to the `BlackoutDates` property. So what happens when a user adds/removes an item from the non-internal instance? Is the UI not updated?

Why can't the `Calendar` just listen to the `CollectionChanged` property on the collection and update itself?